backend/cfg/cfgize.ml

```diff
@@ -569,110 +569,120 @@ let update_trap_handler_blocks : State.t -> Cfg.t -> unit =
-      then
-        Misc.fatal_error
-          "Cfgize.Trap_depth_and_exn.terminator: unexpected handler on self \
-           tailcall"
-      else (stack, exceptional_successor), term
-
-  let process_basic :
-      handler_option ->
-      handler_stack ->
-      Cfg.basic Cfg.instruction ->
-      (handler_stack * handler_option) * Cfg.basic Cfg.instruction =
-   fun exceptional_successor stack instr ->
-    let instr = Cfg.set_trap_depth instr (succ (List.length stack)) in
-    match instr.desc with
-    | Pushtrap { lbl_handler } ->
-      (lbl_handler :: stack, exceptional_successor), instr
-    | Poptrap -> begin
-      match stack with
-      | [] ->
-        Misc.fatal_error
-          "Cfgize.Trap_depth_and_exn.basic: trying to pop from an empty stack"
-      | _ :: stack -> (stack, exceptional_successor), instr
-    end
-    | Op _ | Call _ | Reloadretaddr | Prologue ->
-      ( record_handler stack exceptional_successor
-          ~can_raise:(Cfg.can_raise_basic instr.desc),
-        instr )
+      stack_offset, traps, term
+
+  let rec update_block :
+      Cfg.t -> Label.t -> stack_offset:int -> traps:handler_stack -> unit =
+   fun cfg label ~stack_offset ~traps ->
+    let process_basic :
+        stack_offset:int ->
+        traps:handler_stack ->
+        Cfg.basic Cfg.instruction ->
+        int * handler_stack * Cfg.basic Cfg.instruction =
+     fun ~stack_offset ~traps instr ->
+      let instr = check_and_set_stack_offset instr ~stack_offset ~traps in
+      match instr.desc with
+      | Pushtrap { lbl_handler } ->
+        update_block cfg lbl_handler ~stack_offset ~traps;
+        stack_offset, lbl_handler :: traps, instr
+      | Poptrap -> begin
+        match traps with
+        | [] ->
+          Misc.fatal_error
+            "Cfgize.Trap_depth_and_exn.basic: trying to pop from an empty stack"
+        | _ :: traps -> stack_offset, traps, instr
+      end
+      | Op (Stackoffset n) -> stack_offset + n, traps, instr
+      | Op
+          ( Move | Spill | Reload | Const_int _ | Const_float _ | Const_symbol _
+          | Load _ | Store _ | Intop _ | Intop_imm _ | Negf | Absf | Addf | Subf
+          | Mulf | Divf | Compf _ | Floatofint | Intoffloat | Probe _
+          | Probe_is_enabled _ | Opaque | Begin_region | End_region | Specific _
+          | Name_for_debugger _ )
+      | Call _ | Reloadretaddr | Prologue ->
+        stack_offset, traps, instr
+    in
 
-  let rec update_block : Cfg.t -> Label.t -> handler_stack -> unit =
-   fun cfg label stack ->
     let block = Cfg.get_block_exn cfg label in
     let was_invalid =
-      if block.trap_depth = invalid_trap_depth
+      if block.stack_offset = invalid_stack_offset
       then true
       else begin
-        assert (block.trap_depth = succ (List.length stack));
+        assert (
+          block.stack_offset = compute_stack_offset ~stack_offset ~traps
+          && block.stack_offset >= Proc.trap_size_in_bytes * List.length traps);
```

Comment:
Why is it `>=`?

Reply:
See the answer to a similar question in `linear_to_cfg`, except in cfgize `traps` starts from empty, i.e., has no dummy interproc handler, so no need for `-1`.